src/openforms/prefill/contrib/family_members/tests/test_plugin.py

```diff
@@ -0,0 +1,694 @@
+from pathlib import Path
+from unittest.mock import patch
+
+from django.test import TestCase
+
+import requests_mock
+from freezegun import freeze_time
+from zgw_consumers.test.factories import ServiceFactory
+
+from openforms.authentication.constants import AuthAttribute
+from openforms.config.constants import FamilyMembersDataAPIChoices
+from openforms.config.models import GlobalConfiguration
+from openforms.contrib.haal_centraal.constants import BRPVersions
+from openforms.contrib.haal_centraal.models import HaalCentraalConfig
+from openforms.forms.tests.factories import FormVariableFactory
+from openforms.logging.tests.utils import disable_timelinelog
+from openforms.prefill.service import prefill_variables
+from openforms.submissions.tests.factories import SubmissionFactory
+from openforms.template import render_from_string
+from openforms.utils.tests.vcr import OFVCRMixin
+from openforms.variables.constants import FormVariableDataTypes
+from stuf.constants import EndpointType
+from stuf.stuf_bg.models import StufBGConfig
+from stuf.tests.factories import StufServiceFactory
+
+from ....registry import register
+from ..plugin import PLUGIN_IDENTIFIER
+
+plugin = register[PLUGIN_IDENTIFIER]
+
+
+TEST_FILES = Path(__file__).parent.resolve() / "responses"
+VCR_TEST_FILES = Path(__file__).parent / "files"
+
+
+class FamilyMembersPrefillPluginHCV2Tests(OFVCRMixin, TestCase):
+    """This test case requires the HaalCentraal BRP API to be running.
+    See the relevant Docker compose in the ``docker/`` folder.
+    """
+
+    VCR_TEST_FILES = VCR_TEST_FILES
+
+    def setUp(self):
+        super().setUp()
+
+        hc_config = HaalCentraalConfig(
+            brp_personen_service=ServiceFactory.build(
+                api_root="http://localhost:5010/haalcentraal/api/brp/"
+            ),
+            brp_personen_version=BRPVersions.v20,
+        )
+        hc_config_patcher = patch(
+            "openforms.contrib.haal_centraal.clients.HaalCentraalConfig.get_solo",
+            return_value=hc_config,
+        )
+        global_config = GlobalConfiguration(
+            family_members_data_api=FamilyMembersDataAPIChoices.haal_centraal
+        )
+        global_config_patcher = patch(
+            "openforms.config.models.GlobalConfiguration.get_solo",
+            return_value=global_config,
+        )
+        hc_config_patcher.start()
+        global_config_patcher.start()
+        self.addCleanup(hc_config_patcher.stop)
+        self.addCleanup(global_config_patcher.stop)
+
+    def test_partners_prefill_happy_flow(self):
+        submission = SubmissionFactory.from_components(
+            auth_info__value="999970124",
+            auth_info__attribute=AuthAttribute.bsn,
+            components_list=[],
+        )
+        FormVariableFactory.create(
+            key="hc_prefill_partners_mutable",
+            form=submission.form,
+            data_type=FormVariableDataTypes.array,
+        )
```

Comment:
```suggestion
        FormVariableFactory.create(
            key="hc_prefill_partners_mutable",
            form=submission.form,
            user_defined=True,
            data_type=FormVariableDataTypes.array,
        )
```
please make this representative, we have traits for the common use cases

Reply:
The factory creates a `user-defined` by default that's why I did it like this

Reply:
Ah right, the factory itself is a bit confusing then :P I'll allow it :wink: